sklearn/feature_selection/rfe.py

```diff
@@ -201,16 +251,38 @@ def _fit(self, X, y, step_score=None):
             # for sparse case ranks is matrix
             ranks = np.ravel(ranks)
 
-            # Eliminate the worse features
-            threshold = min(step, np.sum(support_) - n_features_to_select)
+            # Adjust step using special parameters if specified
+            if self.tune_step_at is not None:
```

Comment:
I wonder if this could be neatened by instead implementing the whole thing as:

```
self._main_rfe_loop(X, y, n_features, tune_step_at, self.step)
self._main_rfe_loop(X, y, tune_step_at, n_features_to_select, self.tuning_step)
```
This makes clearer the fact that there are two distinct phases of operation

Reply:
Thank you for the suggestion - will work on it